Python/parking_lot.c

```diff
@@ -67,10 +82,17 @@ _PySemaphore_Init(_PySemaphore *sema)
         Py_FatalError("parking_lot: sem_init failed");
     }
 #else
+    #if defined(CONDATTR_MONOTONIC)
+        if (!condattr_ptr){
+            pthread_condattr_init(&condattr);
+            pthread_condattr_setclock(&condattr, CLOCK_MONOTONIC);
```

Comment:
POSIX says `pthread_condattr_setclock` fails if the clock is unknown. This should be taken care of by the check for `CLOCK_MONOTONIC` but we can still implement a similar check for the return value. I think it would make sense since `CLOCK_MONOTONIC` can be supported at compile time but not runtime.

```
The pthread_condattr_setclock() function may fail if:

[EINVAL]
The value specified by clock_id does not refer to a known clock, or is a CPU-time clock.
These functions shall not return an error code of [EINTR].
```